### PR TITLE
Add `clean` method to `motd` field

### PR DESCRIPTION
**Pull request by PerchunPak**
This will do easier to answer on questions like [this](https://discord.com/channels/936788458939224094/938623566964981802/988019751592820787) or [this](https://discord.com/channels/936788458939224094/938591600160956446/979783704270491659).

I also will port this to #306 when will resolve merge conflict from master.

### CODE REVIEW

**Comment by PerchunPak** [comment]
```py
>>> status = JavaServer.lookup("hypixel.net").status()
>>> status.description
'                §aHypixel Network §c[1.8-1.19]\n            §d§l3RD SKYBLOCK ANNIVERSARY'
>>> status.description.clean()
'                Hypixel Network [1.8-1.19]\n            3RD SKYBLOCK ANNIVERSARY'
```

File: mcstatus/pinger.py
Comment (ItsDrike) [inline mcstatus/pinger.py:37]:
```suggestion
DESCRIPTION_COLORS_RE = re.compile(r"[\xA7|&][0-9A-FK-OR]", re.IGNORECASE)
```
Also change all references to this, and including imports of this constant in other files.

File: mcstatus/pinger.py
Comment (ItsDrike) [inline mcstatus/pinger.py:228]:
```suggestion
            """Get description without color markers (&7, &a, §1 etc.).
```

Also applies for a duplicate class defined in `bedrock_status.py`.

File: mcstatus/bedrock_status.py
Comment (ItsDrike) [inline mcstatus/bedrock_status.py:94]: It may be worth it to make `Description` class in `pinger.py` global, and just import it here directly. I'm not a huge fan of doing the same thing multiple times in different places.

This will become a bit easier after #306 is merged, providing a common place for this class to be in. But right now, let's not make 2 different classes with same logic for the same thing.

**Review by ItsDrike — CHANGES_REQUESTED** [review]

File: mcstatus/pinger.py
Comment (PerchunPak) [reply mcstatus/pinger.py]: Renamed via force push.

File: mcstatus/pinger.py
Comment (PerchunPak) [reply mcstatus/pinger.py]: Changed with force push.

File: mcstatus/pinger.py
Comment (PerchunPak) [reply mcstatus/pinger.py]: Changed with force push.

File: mcstatus/bedrock_status.py
Comment (PerchunPak) [reply mcstatus/bedrock_status.py]: Removed `Motd` class and using `PingResponse.Descriprion` instead now.

File: mcstatus/pinger.py
Comment (ItsDrike) [inline mcstatus/pinger.py:232]:
```suggestion
            return DESCRIPTION_COLORS_RE.sub("", self)
```

**Review by ItsDrike — CHANGES_REQUESTED** [review]
Last small detail and I think this can be merged

File: mcstatus/pinger.py
Comment (PerchunPak) [reply mcstatus/pinger.py]: Done.

**Review by ItsDrike — APPROVED** [review]
Looks good to go, though it may be nice to add some unit-tests before merging into master

**Comment by PerchunPak** [comment]
> Looks good to go, though it may be nice to add some unit-tests before merging into master

I'm sure, regex isn't a something, that should be tested.
I firstly thought about adding parametrize, but this unnecessary. I will add one test just for coverage.

File: mcstatus/tests/test_pinger.py
Comment (ItsDrike) [inline mcstatus/tests/test_pinger.py:395]:
```diff
-    def test_correct_output(self):
+    def test_clean_description(self):
```

**Review by kevinkjt2000 — CHANGES_REQUESTED** [review]
Perhaps we should take this further and make a class dedicated to parsing Minecraft text. Builder methods could support both JSON and section sign (§) formats. We'd have room later to expand the class with outputs for HTML, xt256-color, ANSI, etc.

For now, let's at least avoid putting a class inside of another class.

File: mcstatus/bedrock_status.py
Comment (PerchunPak) [reply mcstatus/bedrock_status.py]: Already discussed and fixed [here](https://github.com/py-mine/mcstatus/pull/329#discussion_r903578388).

**Comment by PerchunPak** [comment]
> Perhaps we should take this further and make a class dedicated to parsing Minecraft text.

As I wrote in discord, this is just fast implement of one feature, that I come up with. I already made #335 for `html`, `xt256`, `ansi` etc. colors and more complex MOTD parsing.

**Comment by PerchunPak** [comment]
So it's approved, maybe merge it?

**Comment by ItsDrike** [comment]
> So it's approved, maybe merge it?

There was a requested change from @kevinkjt2000 which even though you commented on as addressed, kevin didn't yet submit an approving review, dismissing the requested changes review status. I like sticking to a policy of PRs having no requested changes from maintainers before going through with a merge, so even though I approved, I won't merge the PR until kevin makes an approving review too.

**Review by kevinkjt2000 — CHANGES_REQUESTED** [review]
Repeating my earlier comment
> For now, let's at least avoid putting a class inside of another class.

**Comment by kevinkjt2000** [comment]
To be clear, I do see that #335 exists and I still believe that the new code should land in its already planned future home. Why would we add `PingResponse.Description` to overhaul that immediately later, when we could add a new class that is later extensible in #335 for the rest of the planned methods?

**Comment by PerchunPak** [comment]
> To be clear, I do see that #335 exists and I still believe that the new code should land in its already planned future home. Why would we add `PingResponse.Description` to overhaul that immediately later, when we could add a new class that is later extensible in #335 for the rest of the planned methods?

Maybe it should be in #335, **but** #335 is blocked by #306, which is waiting for your review for almost 1.5 months. This feature has also been requested several times on Discord, so should we delay a small (and requested) change for so long?

**Comment by kevinkjt2000** [comment]
> When editing legacy code, follow the legacy structure.

No. By this same logic, I would never do chores. While living in a dirty house, keep living in a dirty house...
We have already moved some other nested classes away from being nested classes. Indeed, we should not add more. There was a lengthy discussion about how nesting classes is unpythonic in the past. `# THIS IS SO UNPYTHONIC` is still in the code in fact.

> which is waiting for your review for almost 1.5 months

I am slow to review, but at the moment I see the do-not-merge label and conflicts that need to be resolved on #306. Personally, I skip over PRs that still have changes outstanding from the checks that I see as an "automatic grader". This way I can optimize my time to review PRs that are ready. Also, this PR has conflicts too.

> so should we delay a small (and requested) change for so long?

Yes, if that feature is not polished enough yet. Users expect quality even if they never ask for it. Open source does not have to be rushed, there is no money on the line here.

---

I am not saying to add all the planned functionality to that class yet. I am suggesting placing the class into its final destination from the start and avoiding rework.

**Comment by PerchunPak** [comment]
> Indeed, we should not add more.

I would like to fix it in specified for this place - #306. It would be worse to put it near `PingResponse` class (instead of in it) and create some chaos in code.

**Comment by kevinkjt2000** [comment]
Let's make the placement correct from the beginning... Why are you insisting on making a mess (that you already plan to cleanup) with the _initial placement_ of this brand new class? I'm saying do that bit of cleanup now. Why wait?

Messy vs. Clean, clean wins.

**Comment by PerchunPak** [comment]
> Why wait?

Because if not follow current style, code will be messy more, than if we would follow the style. Even if it's ugly.

**Comment by PerchunPak** [comment]
However, I will move this PR to #335.

**Comment by kevinkjt2000** [comment]
> oh wait, I did it already in Motd.plain...

Almost as if placing this code into the new class in the correct location was better in the first place... I think you are catching on :)

**Pull request closed without merging**
